mlflow/pyspark/ml/_autolog.py

```diff
@@ -2,7 +2,10 @@
 from functools import reduce
 from typing import Set, Union
 
-from pyspark.errors.exceptions.base import IllegalArgumentException
+try:
+    from pyspark.errors.exceptions.base import IllegalArgumentException
+except ModuleNotFoundError:
+    from pyspark.sql.utils import IllegalArgumentException
```

Comment:
```suggestion
try:
    # ...
    from pyspark.errors.exceptions.base import IllegalArgumentException
except ModuleNotFoundError:
    from pyspark.sql.utils import IllegalArgumentException
```

can we insert a comment?